packages/config-plugins/src/android/BuildGradle.ts

```diff
@@ -0,0 +1,103 @@
+import fs from 'fs-extra';
+import g2js from 'gradle-to-js/lib/parser';
+
+import { getAppBuildGradleFilePath } from './Paths';
+
+// represents gradle command
+// e.g. for `:app:buildExampleDebug` -> { moduleName: app, flavor: example, variant: debug }
+interface GradleCommandInfo {
+  moduleName?: string;
+  flavor?: string;
+  buildType?: string;
+}
+
+interface Config {
+  applicationId?: string;
+  versionCode?: string;
+  versionName?: string;
+}
+
+interface AppBuildGradle {
+  android?: {
+    defaultConfig?: Config;
+    flavorDimensions?: string; // e.g. '"dimension1", "dimension2"'
+    productFlavors?: Record<string, Config>;
+  };
+}
+
+export async function getAppBuildGradleAsync(projectDir: string): Promise<AppBuildGradle> {
+  const buildGradlePath = getAppBuildGradleFilePath(projectDir);
+  const rawBuildGradle = await fs.readFile(buildGradlePath, 'utf8');
+  return await g2js.parseText(rawBuildGradle);
+}
+
+export function resolveConfigValue(
+  buildGradle: AppBuildGradle,
+  flavor: string | undefined,
```

Comment:
```suggestion
  flavor?: string,
```
and make it the last param.